src/chapters/control-flow-and-collections/list.rst

```diff
@@ -0,0 +1,343 @@
+
+In the final lines of the program, we compute the average grade for all
+students:
+
+.. sourcecode:: csharp
+   :lineno-start: 43
+
+   double avg = sum / students.Count;
+   Console.WriteLine("Average grade: " + avg);
+
+List Methods
+-----------------
+
+Let’s gather up a few of the ``List`` methods that we’ve encountered so
+far, along with a few new ones. While these will be the most common methods and
+properties that you use with this class, they by no means represent a complete
+List. Refer to the `official documentation on the List
+class <https://docs.microsoft.com/en-us/dotnet/api/system.collections.List?view=netframework-4.8>`__
+for such a List, and for more details.
+
+To demonstrate the use of these methods, we'll create a new ``List``
+called ``planets``.
+
+.. sourcecode:: csharp
+
+   List<string> planets = new List<string>();
+
+Ok, we've got an empty List. We need to use the class's ``.Add()`` method
+to populate this collection with items.
+
+.. admonition:: Note
+
+   There are other means to declare and initialize an List in fewer lines.
+   These require knowledge of other collections types, so we'll stick with ``.Add()``
+   for the time being.
+
+Using ``.Add()`` to populate ``planets``:
+
+.. sourcecode:: C#
+   :linenos:
+
+   planets.Add("Mercury");
+   planets.Add("Venus");
+   planets.Add("Earth");
+   planets.Add("Mars");
+   planets.Add("Jupiter");
+   planets.Add("Saturn");
+   planets.Add("Uranus");
+   planets.Add("Neptune");
+
+Thus, the first item in this table:
+
+.. _list-methods:
+
+.. list-table:: List methods in C#
+   :header-rows: 1
+
+   * - C# Syntax
+     - Description
+     - Example
+   * - ``Add()``
+     - Adds an item to the List
+     - ``planets.Add("Pluto")`` adds ``Pluto`` to ``planets``
+   * - ``Count``
+     - Returns the number of items in an List, as an ``int``
+     - ``planets.Count`` returns ``9``
+   * - ``Contains()``
+     - Checks to see if the List contains a given item, returning a Boolean
+     - ``planets.Contains("Earth")`` returns ``true``
+   * - ``IndexOf()``
+     - Looks for an item in an List, returns the index of the first occurrence of the item if it exists, returns -1 otherwise
+     - ``planets.IndexOf("Jupiter")`` returns ``4``
+
+Here's a couple more methods that require slightly longer descriptions:
+
+.. _listsort:
+
+.. list-table:: List.Sort()
+   :header-rows: 1
+
+   * - C# Syntax
+     - Description
+     - Example
+   * - ``.Sort()``
+     - Rearranges the elements of an ``List`` into ascending order.
+     - ``planets.Sort()`` produces ``["Earth", "Jupiter", "Mars", "Mercury", "Neptune", "Pluto", "Saturn", "Uranus", "Venus"]``
+
+.. list-table:: ToArray()
+   :header-rows: 1
+
+   * - C# Syntax
+     - Description
+     - Example
+   * - ``ToArray()``
+     - Returns an Array containing the elements of the List
+     - ``planets.ToArray()`` returns
+       ``{"Earth", "Jupiter", "Mars", "Mercury", "Neptune", "Pluto", "Saturn", "Uranus", "Venus"}``
```

Comment:
another c/p holdover maybe? i dont think these need more explanation in C# since sort is a method on list and as far as I can see you dont need to create an array before using toarray on list

Reply:
also, we should be consistent in using ``.`` before. a method or not

Reply:
Reformatted!